src/NzbDrone.Core/Download/Clients/DownloadStation/DownloadStation.cs

```diff
@@ -0,0 +1,302 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using FluentValidation.Results;
+using NLog;
+using NzbDrone.Common.Disk;
+using NzbDrone.Common.Http;
+using NzbDrone.Core.Configuration;
+using NzbDrone.Core.MediaFiles.TorrentInfo;
+using NzbDrone.Core.Parser.Model;
+using NzbDrone.Core.RemotePathMappings;
+using NzbDrone.Common.Extensions;
+using NzbDrone.Core.Validation;
+using System.Net;
+using static NzbDrone.Core.Download.Clients.DownloadStation.DownloadStationTask;
+using System.IO;
+
+namespace NzbDrone.Core.Download.Clients.DownloadStation
+{
+    public class DownloadStation : TorrentClientBase<DownloadStationSettings>
+    {
+        protected readonly IDownloadStationProxy _proxy;
+        protected static readonly Dictionary<DownloadStationTaskStatus, DownloadItemStatus> StatusTable = new Dictionary<DownloadStationTaskStatus, DownloadItemStatus>();
+
+        static DownloadStation()
+        {
+            StatusTable.Add(DownloadStationTask.DownloadStationTaskStatus.Waiting, DownloadItemStatus.Queued);
+            StatusTable.Add(DownloadStationTask.DownloadStationTaskStatus.Error, DownloadItemStatus.Failed);
+            StatusTable.Add(DownloadStationTask.DownloadStationTaskStatus.Paused, DownloadItemStatus.Paused);
+            StatusTable.Add(DownloadStationTask.DownloadStationTaskStatus.Seeding, DownloadItemStatus.Completed);
+        }
+
+        public DownloadStation(IDownloadStationProxy proxy,
+                     ITorrentFileInfoReader torrentFileInfoReader,
+                     IHttpClient httpClient,
+                     IConfigService configService,
+                     IDiskProvider diskProvider,
+                     IRemotePathMappingService remotePathMappingService,
+                     Logger logger)
+            : base(torrentFileInfoReader, httpClient, configService, diskProvider, remotePathMappingService, logger)
+        {
+            this._proxy = proxy;
+        }
+
+        public override string Name
+        {
+            get
+            {
+                return "Download Station";
+            }
+        }
+
+        protected virtual OsPath GetOutputPath(OsPath outputPath, DownloadStationTask torrent)
+        {
+            return outputPath + torrent.Title;
+        }
+
+        public override IEnumerable<DownloadClientItem> GetItems()
+        {
+            IEnumerable<DownloadStationTask> torrents;
+
+            try
+            {
+                torrents = _proxy.GetItems(Settings);
+            }
+            catch (DownloadClientException ex)
+            {
+                _logger.Error(ex, ex.Message);
+                return Enumerable.Empty<DownloadClientItem>();
+            }
+
+            var items = new List<DownloadClientItem>();
+
+            foreach (var torrent in torrents)
+            {
+                var outputPath = new OsPath(torrent.Additional.Detail["destination"]);
```

Comment:
Done. Caching is working with no expiration date because I don't know how to estimate a good number.

 Now to the remote mapping part.

Possible scenarios:

Sonnar running on a Diskstation
	I guess SharedFolderResolver is good enough here because it will always resolve to a phisical path that sonarr has access to.
	
Sonnar not running on a Diskstation
	Here we need to configure RemoteMapping, the question is with which one? The shared folder or the physical one? I'm voting for shared because it's easier for the user to remember.

What do you think of this, am i missing something?